apps/web/pages/api/intent-username/index.ts

```diff
@@ -0,0 +1,51 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { getSession } from "next-auth/react";
+
+import prisma from "@calcom/prisma";
+import { Prisma } from "@calcom/prisma/client";
+
+import { checkUsername } from "@lib/core/server/checkUsername";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { intentUsername } = req.body;
```

Comment:
1. We're not checking for http method in this endpoint
2. [Prefer defaultHandler for simple API handlers](https://app.gitbook.com/o/6snd8PyPYMhg0wUw6CeQ/s/VXRprBTuMlihk37NQgUU/engineering/best-practices/prefer-defaulthandler-for-simple-api-handlers)